presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedClusterStatsResource.java

```diff
@@ -16,60 +16,90 @@
+import com.facebook.presto.metadata.AllNodes;
 import com.facebook.presto.resourceGroups.FileResourceGroupConfigurationManagerFactory;
 import com.facebook.presto.server.ClusterStatsResource;
 import com.facebook.presto.server.testing.TestingPrestoServer;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
 import com.facebook.presto.tests.DistributedQueryRunner;
 import com.google.common.collect.ImmutableMap;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
+import java.util.Map;
 import java.util.Optional;
+import java.util.concurrent.TimeoutException;
 
 import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
 import static com.facebook.airlift.http.client.JsonResponseHandler.createJsonResponseHandler;
 import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
 import static com.facebook.airlift.json.JsonCodec.jsonCodec;
 import static com.facebook.airlift.testing.Closeables.closeQuietly;
 import static com.facebook.presto.client.PrestoHeaders.PRESTO_USER;
+import static com.facebook.presto.execution.QueryState.QUEUED;
+import static com.facebook.presto.execution.QueryState.RUNNING;
 import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunner;
-import static com.facebook.presto.utils.QueryExecutionClientUtil.runToExecuting;
+import static com.facebook.presto.utils.QueryExecutionClientUtil.runToFirstResult;
+import static com.facebook.presto.utils.QueryExecutionClientUtil.runToQueued;
 import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
 import static java.lang.Thread.sleep;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertNotNull;
 import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
 
 public class TestDistributedClusterStatsResource
 {
+    private static final String RESOURCE_GROUPS_CONFIG_FILE = "resource_groups_config_simple.json";
+    private static final int COORDINATOR_COUNT = 2;
+    private static final String RESOURCE_GROUP_GLOBAL = "global";
     private HttpClient client;
-    private TestingPrestoServer coordinator;
+    private TestingPrestoServer coordinator1;
     private TestingPrestoServer resourceManager;
+    private TestingPrestoServer coordinator2;
 
     @BeforeClass
     public void setup()
             throws Exception
     {
         client = new JettyHttpClient();
-        DistributedQueryRunner runner = createQueryRunner(ImmutableMap.of("query.client.timeout", "10s"), 2);
-        coordinator = runner.getCoordinator(0);
+        DistributedQueryRunner runner = createQueryRunner(
+                ImmutableMap.of(
+                        "resource-manager.query-expiration-timeout", "4m",
+                        "resource-manager.completed-query-expiration-timeout", "4m"),
```

Comment:
Given the sequence Id fix has been merged, do we still want this timeout to be set for the test?

Reply:
I rebased and re-run without these configs for expiration-timeout, the test fails. I think there could be other issues apart from the issue addressed by sequence Id fix.